test/integration/user.js

```diff
@@ -3,57 +3,151 @@ import path from 'path';
+              chai.assert.equal(resp.status, 200, resp.result);
+              chai.assert.isObject(resp.result);
 
-          done();
-        })
-        .catch(done);
-    });
-  });
+              chai.assert.isString(resp.result.name);
+              chai.assert.isDefined(resp.result.fileBinary);
+
+              done();
+            })
+            .catch(done);
+        });
+      });
 
-  describe('upload', () => {
-    it('upload request is successful', (done) => {
-      fs.readFile(path.resolve('test/fixtures/test.txt'), 'utf8')
-        .then((contents) => {
-          dbx.filesUpload({ path: '/test.txt', contents })
+      describe('upload', () => {
+        it('upload request is successful', (done) => {
+          fs.readFile(path.resolve('test/fixtures/test.txt'), 'utf8')
+            .then((contents) => {
+              dbx.filesUpload({ path: '/test.txt', contents })
+                .then((resp) => {
+                  chai.assert.instanceOf(resp, DropboxResponse);
+                  chai.assert.equal(resp.status, 200, resp.result);
+                  chai.assert.isObject(resp.result);
+
+                  done();
+                })
+                .catch(done);
+            });
+        });
+      });
+
+      describe('app auth', () => {
+        it('successfully uses app auth', (done) => {
+          dbx.checkApp({ query: 'Echo string' })
+            .then((resp) => {
+              chai.assert.instanceOf(resp, DropboxResponse);
+              chai.assert.equal(resp.status, 200, resp.result);
+              chai.assert.isObject(resp.result);
+              chai.assert.equal('Echo string', resp.result.result);
+
+              done();
+            })
+            .catch(done);
+        });
+      });
+
+      describe('token refresh', () => {
+        it('refreshes the token when necessary', (done) => {
+          const currentDate = new Date();
+          const dbxAuthWithRefreshToken = new DropboxAuth({
+            accessToken: appInfo[appType].token,
+            refreshToken: appInfo[appType].refresh_token,
+            clientId: appInfo[appType].app_key,
+            clientSecret: appInfo[appType].app_secret,
+            accessTokenExpiresAt: currentDate.setHours(currentDate.getHours() - 20),
+          });
+
+          const dbxWithRefreshToken = new Dropbox({ auth: dbxAuthWithRefreshToken });
+
+          dbxWithRefreshToken.usersGetCurrentAccount()
             .then((resp) => {
               chai.assert.instanceOf(resp, DropboxResponse);
               chai.assert.equal(resp.status, 200, resp.result);
               chai.assert.isObject(resp.result);
+              chai.assert.notEqual(
+                dbxAuthWithRefreshToken.getAccessToken(), appInfo[appType].token,
+              );
 
               done();
             })
             .catch(done);
         });
+      });
     });
+  }
+}
+
+describe('incorrect auth', () => {
+  it('fails if there is no auth object', () => {
```

Comment:
this might be named a little weird; this would be an empty auth object not no auth object right?